src/test/java/io/fabric8/maven/docker/util/AuthConfigFactoryTest.java

```diff
@@ -112,6 +112,35 @@ public void testSystemProperty() throws Exception {
         }
     }
 
+    @Test
+    public void testDockerSystemProperty() throws Exception {
+        System.setProperty("docker.username","roland");
+        System.setProperty("docker.password", "secret");
+        System.setProperty("docker.email", "roland@jolokia.org");
+        try {
+            AuthConfig config = factory.createAuthConfig(true, false, null, settings, null, null);
+            verifyAuthConfig(config,"roland","secret","roland@jolokia.org");
+        } finally {
+            System.clearProperty("docker.username");
+            System.clearProperty("docker.password");
+            System.clearProperty("docker.email");
+        }
+    }
+
+    @Test
+    public void testRegistrySystemProperty() throws Exception {
```

Comment:
can we please avoid copy & pasting the test ? Just extract a common method which is called once with "docker" and once with "registry" from each individual test.

Helpful would be also a test that ensures that "docker" has precedence over "registry" to keep the default behaviour.